Create Blog “how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise”

GuopingJia opened the pull request:
Automatically generated by Netlify CMS

choukden commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
````diff
@@ -0,0 +1,660 @@
+The Kasten dashboard will be available at: `http://127.0.0.1:8080/k10/#/`
+```
+
+With above commands, Kasten K10 is installed to the namespace *'kasten-io'* in the cluster. To validate the installation, typing the following command to watch for the status of all Pods. Helm installs a list of Pods to the namespace. It takes a while before all those Pods start running.  
````
Second sentence:
typing the following command to watch the status of all Pods.  (remove the "for").

choukden reviewed (changes requested):
Hi Guoping,

Thanks again for your new blog post. It is a great post.

Maybe, I would suggest to cross link the blog post with your previous blog post.

Something like that:

In the second paragraph of the introductory section for this blog post you may want to add something like that:

In the blog post [Getting started with volume snapshot on K8s cluster](https://developer.hpe.com/blog/getting-started-with-volume-snapshots-on-a-kubernetes-cluster-in-hpe-greenlake-for-private-cloud-enterprise/), I explained you how to create a volume snapshot of a MySQL database running on a K8s cluster deployed on HPE GreenLake for Private Cloud Enterprise.

In this post I will show you how to backup and restore the stateful applications deployed in a K8s cluster in HPE GreenLake....

**Then**, in your previous blog post "Getting started with volume snapshots on a Kubernetes cluster in HPE GreenLake for Private Cloud Enterprise", in the **Summary** section at the end of the post, you may want to cross link your new blog post. Something like that:
 
I﻿n this blog post, I described persistent volumes, volume snapshots, and the CSI driver in K8s. Using HPE CSI driver for K8s, I demonstrated how to create a volume snapshot of a MySQL database and how to restore database using the created volume snapshot in the cluster. The volume snapshot capability can be easily integrated with third-party tools like Kasten K10 by Veeam as an automatic backup and recovery solution. It can significantly simplify the process and enhance the robustness of data management in a K8s cluster. Feel free to take a look at my blog post [How to backup and restore stateful app using Kasten 10](https://developer.hpe.com/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise/).

Thanks again
Denis

GuopingJia commented:
> Hi Guoping,
> 
> Thanks again for your new blog post. It is a great post.
> 
> Maybe, I would suggest to cross link the blog post with your previous blog post.
> 
> Something like that:
> 
> In the second paragraph of the introductory section for this blog post you may want to add something like that:
> 
> In the blog post [Getting started with volume snapshot on K8s cluster](https://developer.hpe.com/blog/getting-started-with-volume-snapshots-on-a-kubernetes-cluster-in-hpe-greenlake-for-private-cloud-enterprise/), I explained you how to create a volume snapshot of a MySQL database running on a K8s cluster deployed on HPE GreenLake for Private Cloud Enterprise.
> 
> In this post I will show you how to backup and restore the stateful applications deployed in a K8s cluster in HPE GreenLake....
> 
> **Then**, in your previous blog post "Getting started with volume snapshots on a Kubernetes cluster in HPE GreenLake for Private Cloud Enterprise", in the **Summary** section at the end of the post, you may want to cross link your new blog post. Something like that:
> 
> I﻿n this blog post, I described persistent volumes, volume snapshots, and the CSI driver in K8s. Using HPE CSI driver for K8s, I demonstrated how to create a volume snapshot of a MySQL database and how to restore database using the created volume snapshot in the cluster. The volume snapshot capability can be easily integrated with third-party tools like Kasten K10 by Veeam as an automatic backup and recovery solution. It can significantly simplify the process and enhance the robustness of data management in a K8s cluster. Feel free to take a look at my blog post [How to backup and restore stateful app using Kasten 10](https://developer.hpe.com/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise/).
> 
> Thanks again Denis

Thanks Denis !

This is a very good suggestion. I was thinking about the same cross-linking of those 2 blog posts. I will get them updated tomorrow morning when I get in. 

An extremely busy day, working on quite a few issues in one customer engagement. Running out of the energy for today.

Guoping

GuopingJia commented:
Hi, Denis, 

I've updated 2 blog posts and added the cross links.

Those 2 blog posts may need to be merged at the same time to avoid links being broken. 

Thank you for all your help !

Guoping

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
````diff
@@ -0,0 +1,660 @@
+replicaset.apps/snapshot-controller-5fd799f6b5   2         2         2       56d
+```
+
+HPE CSI driver for K8s supports both dynamical persistent volumes and volume snapshots. The following are the *StorageClasses* and the *VolumeSnapshotClass* that are configured in the cluster: 
````
I think you mean "dynamic", not "dynamical" (sounds kind of demonic... lol)

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
````diff
@@ -0,0 +1,660 @@
+gl-sbp-frank-gl1-sstor01             csi.hpe.com   Delete           56d
+```
+
+[The joint partnership between HPE and Veeam](https://www.kasten.io/kubernetes/resources/blog/kubernetes-backup-with-hpe-csi-and-kasten-k10) supports HPE CSI driver for K8s and Kasten K10 as a data management solution for K8s backup and recovery. The following sections will show you how to install Kasten K10 on the cluster and how to use it with HPE CSI driver for K8s to backup and restore the pe ...
````
add the word "the" prior to "HPE CSI driver for K8s to backup and restore..."

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
```diff
@@ -0,0 +1,660 @@
+
+### Prerequisites
+
+Before starting, make sure you meet the following requirements:
```
Similar to the other blog post, I'd recommend rephrasing this to:

"Before starting, make sure you have the following:"

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
````diff
@@ -0,0 +1,660 @@
+The Kasten dashboard will be available at: `http://127.0.0.1:8080/k10/#/`
+```
+
+With above commands, Kasten K10 is installed to the namespace *'kasten-io'* in the cluster. To validate the installation, typing the following command to watch the status of all Pods. Helm installs a list of Pods to the namespace. It takes a while before all those Pods start running.  
````
Add the word "the" prior to "above commands" so it reads, "With the above commands, Kasten..."

In the second sentence, change "typing" to "type" so it reads "To validate the installation, type the following command..."

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
````diff
@@ -0,0 +1,660 @@
+service/gateway edited
+```
+
+T﻿yping the following command to get the *gateway* service endpoint:
````
Change to "Type the following command..."

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
````diff
@@ -0,0 +1,660 @@
+gl-tor-upc-cp-gw-node1.customer.yyz.gl-hpe.local:10021
+```
+
+T﻿he Kasten K10 service dashboard can then be accessed by pointing your browser to the URL '*http://gl-tor-upc-cp-gw-node1.customer.yyz.gl-hpe.local:10021/k10/#/'* :
````
I would change "then" to "now", so it's "can now be accessed by...

Rational: you are giving instructions to the reader so they are doing it in the moment, not in the future or past.

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
```diff
@@ -0,0 +1,660 @@
+
+![](/img/k10-login.png)
+
+C﻿lick *Accept Terms* after specifying your email and company name, you will be landed to Kasten K10 Dashboard:
```
Change to:
Click Accept Terms after specifying your email and company name. This will land you on the Kasten K10 dashboard."

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
```diff
@@ -0,0 +1,660 @@
+
+![](/img/k10-dashboard.png)
+
+Kasten K10 automatically discovers all the applications and their data across namespaces in the cluster. The K10 Dashboard displays a list of applications that are mapped to namespaces. It also displays a summary of the cluster’s backup data footprint, showing *0.0 B* when accessing the dashboard for the first time. 
```
I'm pretty sure that "Dashboard" in this instance isn't actually part of the name (which would make it a proper noun). It's just a dashboard that's being designated as a dashboard that's specific to K10. So, in this case, it should have a lower case initial letter - "dashboard".

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
```diff
@@ -0,0 +1,660 @@
+
+Whenever Kasten K10 detects volumes that were provisioned via the CSI driver deployed in the cluster, it will look for a *VolumeSnapshotClass* with this K10 annotation for the identified CSI driver and use it to create snapshots. 
+
+Typing the following command to verify the *VolumeSnapshotClass* has the required K10 annotation added:
```
Change "Typing" to "Type"

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
````diff
@@ -0,0 +1,660 @@
+  - mysql-pvc.yaml
+```
+
+T﻿yping below command to install the MySQL database to the namespace *'mysql'*:
````
Change this to "Type the command shown below to install the My SQL..."

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
````diff
@@ -0,0 +1,660 @@
+deployment.apps/mysql created
+```
+
+T﻿yping the following command to check the MySQL database deployment state. The MySQL Pod should be in *Running* status:
````
Change to "Type the following command..."

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
````diff
@@ -0,0 +1,660 @@
+replicaset.apps/mysql-6974b58d48   1         1         1       24s
+```
+
+Y﻿ou can check the *PersistentVolume* (PV) and the PVC get provisioned as part of the MySQL database deployment:
````
Modify to "You can check **that** the Persistent..."

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
```diff
@@ -0,0 +1,660 @@
+
+**2. Access MySQL database**
+
+In order to access MySQL database service using the mysql CLI, set first the port-forward of *service/mysql*:   
```
Change the second phrase in the sentence to "...CLI, first set the port-forward..." (swapping the position of the words "first" and "set"

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
```diff
@@ -0,0 +1,660 @@
+
+**3. Populate MySQL database** 
+
+The MySQL application repo has the *'test'* folder that contains a list of scripts for populating data records and testing the contents: 
```
Change "the test folder" to "a test folder"

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
````diff
@@ -0,0 +1,660 @@
+└── test_employees_sha.sql
+```
+
+Typing the following command to populate a sample *employees* data to the MySQL database:
````
Change "Typing" to "Type"

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
````diff
@@ -0,0 +1,660 @@
+NULL
+```
+
+T﻿he added sample data records *employees* can be checked and verified by running below commands:
````
Modify to "The added sample data records **called** employees can be checked and verified by **running the commands shown below:"**

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
```diff
@@ -0,0 +1,660 @@
+
+### Back up MySQL database
+
+In order to back up the MySQL database, go to the Kasten K10 Dashboard and click the *Applications*. Find the deployed MySQL database *'mysql'* from the application list and expand its menu. Then click *Snapshot* button. 
```
Change "Kasten K10 Dashboard" to "Kasten K10 dashboard"

Add the word "the" prior to "Snapshot" in the last sentence.

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
```diff
@@ -0,0 +1,660 @@
+
+![](/img/k10-backup.png)
+
+T﻿he snapshot of the MySQL database will be started that takes a few seconds. Go back to the K10 Dashboard, you should see the completed *Backup* entry under **Actions** with protected object as *mysql*:
```
Change first few sentences to read:
The snapshot of the MySQL database will be started. This takes a few seconds. When you go back to the K10 dashboard, you should see..."

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
```diff
@@ -0,0 +1,660 @@
+
+![](/img/k10-data-backup.png)
+
+I﻿n the cluster, after snapshot of the MySQL database, you can check there is a *VolumeSnapshot* *'k10-csi-snap-ltxzrwxgp6r5pwkp'* created f﻿rom the source PVC *'mysql-pvc'* in the namespace *mysql*, together with a *VolumeSnapshotContent* object created at cluster level. The *READYTOUSE* of the *VolumeSnapshot* should be showing as *true*:
```
Modify part of the first sentence to:
"...you can check **that** there is a..."

In the last sentence, consider rephrasing to "...should show as true."

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
```diff
@@ -0,0 +1,660 @@
+
+### Restore MySQL database
+
+B﻿efore showing the database restore, I﻿ will first delete some table from MySQL database to simulate a loss of data. Then, I will perform the database recovery using the Kasten K10.
```
change "some" to "a", so it reads "I will first delete a table from..."

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
````diff
@@ -0,0 +1,660 @@
+Query OK, 9 rows affected (1,523 sec)
+```
+
+I﻿f re-run the testing script *test_employees_sha.sql*, it will show the failures of *CRC* and *count* which indicate the loss of data in the MySQL database:
````
Modify to "If you re-run the testing script..."
Place a comma after "count" 
If you rerun the testing script... it will show the failures of CRC and count, which indicate the loss of data...

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
```diff
@@ -0,0 +1,660 @@
+
+**2. Perform MySQL database restore**
+
+In order to restore the MySQL database, g﻿o to the Kasten K10 Dashboard, locate the MySQL database *'mysql'* from the application list, expand the menu of *mysql*, then click *Restore* button: 
```
Change Dashboard to dashboard
Place the word "and" prior to "then click..."

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
```diff
@@ -0,0 +1,660 @@
+
+![](/img/k10-restore-point.png)
+
+U﻿se all the default options from **Restore Point** and click *Restore* button:
```
Add the word "the" prior to "Restore button:"

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
```diff
@@ -0,0 +1,660 @@
+
+![](/img/k10-restore.png)
+
+T﻿he restore of the MySQL database will be started from the selected restore point. It will take a few seconds. Go back to the Kasten K10 Dashboard, you should see the completed *Restore* entry under **Actions** with target namespace as *mysql*:
```
Dashboard should be dashboard.
Modify the third sentence and break it out so it reads like this:
Go back to the Kasten K10 dashboard. You should see the completed..."

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
````diff
@@ -0,0 +1,660 @@
++---------+--------+
+```
+
+T﻿his indicates the MySQL database gets recovered from its backup and MySQL database data is back !
````
Modify sentence to read:
"This indicates that the MySQL database has been recovered from its backup and the MySQL database data is back!" (I also noticed that there's currently an extra space prior to the exclamation point.)

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
```diff
@@ -0,0 +1,660 @@
+
+### Summary
+
+In this blog post, I explored the functionalities of Kasten K10 and HPE CSI driver for K8s. Using the volume snapshot capability in HPE CSI driver for K8s, I demonstrated how to use Kasten K10 to backup the persistent volume of a sample MySQL database deployed in the cluster in HPE GreenLake for Private Cloud Enterprise. I then illustrated how to restore database from the backup. Kasten K10, with ...
```
In the third sentence, modify slightly to:
"I then illustrated how to restore **the** database from the backup."

drensing commented on content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md:
```diff
@@ -0,0 +1,660 @@
+
+In this blog post, I explored the functionalities of Kasten K10 and HPE CSI driver for K8s. Using the volume snapshot capability in HPE CSI driver for K8s, I demonstrated how to use Kasten K10 to backup the persistent volume of a sample MySQL database deployed in the cluster in HPE GreenLake for Private Cloud Enterprise. I then illustrated how to restore database from the backup. Kasten K10, with ...
+
+You can keep coming back to the [HPE Developer blog](https://developer.hpe.com/blog/) to learn more about HPE GreenLake for Private Cloud Enterprise.
```
Modify to "Please keep coming back...."

drensing reviewed (commented):
Excellent post! I don't know how to configure this, but I'm pretty sure, if I followed your instructions, I'd be successful!

GuopingJia commented:
> Excellent post! I don't know how to configure this, but I'm pretty sure, if I followed your instructions, I'd be successful!

Thanks Dale for all your comments. I really enjoy updating the blog post with your comments/good suggestions!

The new patch should contain all the changes to your comments. Let me know if there is anything you want me to change. 

Thank you !

Guoping
